pkg/api/types.go

```diff
@@ -1359,6 +1359,11 @@ type ProjectDirectoryImageBuildInputs struct {
 	// that will populate the build context for the Dockerfile or
 	// alter the input image for a multi-stage build.
 	Inputs map[string]ImageBuildInputs `json:"inputs,omitempty"`
+
+	// CustomBuildStrategy indicates that the image should be executed as
+	// a custom build strategy insttead of a Docker Build Strategy (default).
+	// The image must support the Custom Build API and emit an OCI Container.
+	CustomBuildStrategy bool `json:"custom_build_strategy,omitempty"`
```

Comment:
Could we expose something more like

```go
type ProjectDirectoryImageBuildInputs struct {
  // ...
  BuildStrategyType buildapi.BuildStrategyType `json:"build_strategy_type,omitempty"
  
  BuildOptions *BuildOptions `json:"build_options,omitempty"
}

type BuildOptions struct {
  Docker *DockerOptions `json:"docker,omitempty"`
  Custom *CustomOptions `json:"custom,omitempty"`
}

type CustomOptions struct {
  ServiceAccountRoles []string `json:"serviceaccount_roles,omitempty"`
  Secrets []buildapi.SecretSpec `json:"secrets,omitempty"`
}
```

Then we can do validation that e.g. people don't set the context dir/dockerfile path when asking for a custom strategy, etc.

Reply:
We can probably just close this PR - it's going to take a bit to figure out the interface model and how to do this in an acceptable manner. No point in having this PR continue its decline into staleness.